source/common/http/header_map_impl.cc

```diff
@@ -86,9 +91,11 @@ void HeaderString::append(const char* data, uint32_t size) {
     // Rather than be too clever and optimize this uncommon case, we dynamically
     // allocate and copy.
     type_ = Type::Dynamic;
-    dynamic_capacity_ =
-        std::max(MinDynamicCapacity, static_cast<size_t>((string_length_ + size) * 2));
-    validateCapacity(dynamic_capacity_);
+    const uint64_t new_capacity = newCapacity(string_length_, size);
+    dynamic_capacity_ = std::max(MinDynamicCapacity, new_capacity);
+    if (dynamic_capacity_ != MinDynamicCapacity) {
+      validateCapacity(new_capacity);
```

Comment:
For some reason, my comment was lost here. I was wondering if we could move `validateCapacity` up and make it unconditional?